WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationTracker.kt

```diff
@@ -4,9 +4,114 @@ import org.wordpress.android.analytics.AnalyticsTracker
+                mapOf(
+                        "segment_name" to segmentName,
+                        "segment_id" to segmentId
+                )
+        )
+    }
+
+    fun trackVerticalsViewed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_VIEWED)
+    }
+
+    fun trackVerticalSelected(verticalName: String, verticalId: String, isUserVertical: Boolean) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_SELECTED,
+                mapOf(
+                        "vertical_name" to verticalName,
+                        "vertical_id" to verticalId,
+                        "vertical_is_user" to isUserVertical
+                )
+        )
+    }
+
+    fun trackVerticalsSkipped() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_SKIPPED)
+    }
+
+    fun trackBasicInformationViewed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_VIEWED)
+    }
+
+    fun trackBasicInformationCompleted(siteTitle: String, siteTagLine: String) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_COMPLETED,
+                mapOf(
+                        "site_title" to siteTitle,
+                        "tagline" to siteTagLine
+                )
+        )
+    }
+
+    fun trackBasicInformationSkipped() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_SKIPPED)
+    }
+
+    fun trackDomainsAccessed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_DOMAINS_ACCESSED)
+    }
+
+    fun trackDomainSelected(choosenDomain: String, searchTerm: String) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_DOMAINS_SELECTED,
+                mapOf(
+                        "chosen_domain" to choosenDomain,
+                        "search_term" to searchTerm
+                )
+        )
+    }
+
+    fun trackPreviewLoading() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SUCCESS_LOADING)
+    }
+
+    fun trackPreviewWebviewShown() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SUCCESS_PREVIEW_VIEWED)
+    }
+
+    fun trackPreviewWebviewFullyLoaded() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SUCCESS_PREVIEW_LOADED)
+    }
+
+    fun trackCreationCompleted() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_COMPLETED)
+    }
+
+    fun trackConnectionErrorShown(origin: String) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_ERROR_SHOWN,
+                mapOf("error_info" to "connection error - $origin")
+        )
+    }
+
+    fun trackGenericErrorShown(origin: String) {
+        AnalyticsTracker.track(
```

Comment:
Should we use the below signature of the `AnalyticsTracker.track` instead?

```
public static void track(Stat stat, String errorContext, String errorType, String errorDescription) {
```

I think it'd be worth adding the error type to the tracker and it'd be great to be consistent with other error tracking if it's at all possible.

Reply:
If we end up making a change here, I think it'd be worth renaming this method to something like `trackNetworkErrorShown` since we want to track all network errors, not just generic ones.

Reply:
Good point! I overlooked the method for error tracking.

Fixed in [#a9d93d5](https://github.com/wordpress-mobile/WordPress-Android/pull/8998/commits/a9d93d5474eea831bab5064e9a76dd199f67a0a3)

Reply:
Btw I didn't use `java.class.simpleName` for errorContext as I think it's not a good idea to tie class name to tracking. It might cause confusion if we ever decide to rename the class.

Reply:
Sure, it's up to you. We may want a bit more structure around that though, see my second review for this. (not posted yet)